src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+            {"developer"}};
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReservedDeploymentExistence(String mkType) {
+
+        String namespace = String.format("reserved-kafka-%s-1", mkType);
+        log.info("evaluating namespace '{}'", namespace);
+
+        String managedKafkaName = String.format("reserved-kafka-%s-1", mkType);
+        log.info("managed kafka reserved deployment '{}'", managedKafkaName);
+        Optional<ManagedKafka> mkOptional = FleetshardUtils.managedKafka(oc).inNamespace(namespace).list().getItems().stream().filter(ManagedKafka::isReserveDeployment).findAny();
+        Assert.assertTrue(mkOptional.isPresent(), "reserved deployment is not present");
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    public void testReservedDeploymentDoesNotPreventKafkaCreation(String mkType) throws Exception {
+
+        log.info("testing managed kafka type: '{}'", mkType);
+
+        // obtain max limit of kafka instances per given mk type (developer, standard)
+        int upperLimitPerManagedKafkaType;
+        switch (mkType) {
+            case "standard":
+                upperLimitPerManagedKafkaType = maxStandardInstancesInCluster;
+                break;
+            case "developer":
+                upperLimitPerManagedKafkaType = maxDeveloperInstancesInCluster;
+                break;
+            default:
+                throw new Exception("Unsupported managed kafka type");
+        }
+        log.info("upper limit for instance type '{}' is '{}' instances.", mkType, upperLimitPerManagedKafkaType);
+
+        int currentInstancesCount = FleetshardUtils.getCountOfExistingGivenManagedKafkaCRType(oc, ManagedKafkaType.valueOf(mkType));
+        log.info("currently there are '{}' instances of given type", currentInstancesCount);
+
+        // if there are already max number of instances skip test
+        if (currentInstancesCount > upperLimitPerManagedKafkaType) {
+            log.warn("currently too many instances ({}/{}), of type '{}'",
+                currentInstancesCount,
+                upperLimitPerManagedKafkaType,
+                mkType);
+            throw new SkipException("Too many existing instances, which would cause quota to be breached");
+        }
+
+        // create kafka instance of expected type
+        KafkaRequestPayload payload = new KafkaRequestPayload()
+            .name(KAFKA_INSTANCE_NAME)
+            .cloudProvider("aws")
+            .region("us-east-1")
+            .plan(String.format("%s.x1", mkType));
+
+        log.info("attempt to create kafka instance");
+        KafkaRequest kafkaRequest = null;
+        try {
+            kafkaRequest = kafkaMgmtApi.createKafka(true, payload);
+            log.debug(kafkaRequest);
+            log.info("wait for provisioning of kafka instance with id '{}'", kafkaRequest.getId());
+            KafkaMgmtApiUtils.waitUntilKafkaIsProvisioning(kafkaMgmtApi, kafkaRequest.getId());
```

Comment:
Looking at the implementation of `waitUntilKafkaIsProvisioning` https://github.com/bf2fc6cc711aee1a0c2a/e2e-test-suite/blob/04c43345024f9dff671221c74a9422b6f5360ca2/src/main/java/io/managed/services/test/client/kafkamgmt/KafkaMgmtApiUtils.java#L308 I think the test could pass for wrong reasons (e.g the kafka is in `preparing` state, `provisioning` state but not already sent to the data plane etc).

The wait condition should be that the kafka is in `ready` state. And for that I think the util https://github.com/bf2fc6cc711aee1a0c2a/e2e-test-suite/blob/04c43345024f9dff671221c74a9422b6f5360ca2/src/main/java/io/managed/services/test/client/kafkamgmt/KafkaMgmtApiUtils.java#L397  is more appropriate

Reply:
yes waiting for provisioning state was due to not needing to wait for actual creation. (just question of time), 
I assumed that preparing is good enough condition.

Reply:
> yes waiting for provisioning state was due to not needing to wait for actual creation. (just question of time),
I assumed that preparing is good enough condition.

No, we actually need to be sure that Kafka is successfully provisioned in the data plane i.e it's status is `ready`.

Reply:
Okay thanks. Just note: I still see it as waiting for provisioning, so it could be that the change is not pushed yet ;)